buildpack/dockerfile.go

```diff
@@ -32,13 +32,22 @@ const (
 	warnCommandNotRecommended           = "%s command %s on line %d is not recommended"
 )
 
-var permittedCommands = []string{"FROM", "ADD", "ARG", "COPY", "ENV", "LABEL", "RUN", "SHELL", "USER", "WORKDIR"}
+var recommendedCommands = []string{"FROM", "ADD", "ARG", "COPY", "ENV", "LABEL", "RUN", "SHELL", "USER", "WORKDIR"}
 
 type DockerfileInfo struct {
 	ExtensionID string
 	Kind        string
 	Path        string
-	NewBase     string
+	// WithBase if populated indicates that the Dockerfile switches the image base to the provided value.
+	// If WithBase is empty, Extend should be true, otherwise there is nothing for the Dockerfile to do.
+	// However if WithBase is populated, Extend may be true or false.
+	WithBase string
```

Comment:
For some reason `WithBase` seemed more clear than `NewBase`. But we can change it back if that's not true

Reply:
https://en.wikipedia.org/wiki/Ace_of_Base